modules/library/main/src/main/java/org/geotools/filter/function/EqualIntervalFunction.java

```diff
@@ -220,4 +242,53 @@ public RangedClassifier evaluate(Object object) {
         }
         return calculate((SimpleFeatureCollection) object);
     }
+
+    private double[] getNotNumericalPercentages(int classNum, int totalSize) {
+        int lastBigBin = totalSize % classNum;
+        if (lastBigBin != 0) lastBigBin--;
+        double classMembers = (double) totalSize / classNum;
+        double[] percentages = new double[classNum];
+        for (int i = 0; i < classNum; i++) {
+            percentages[i] = (classMembers / totalSize) * 100;
+            if (lastBigBin != 0 && lastBigBin == i) {
+                classMembers--;
+            }
+        }
+        return percentages;
+    }
+
+    private double[] getNumericalPercentages(
+            int classNum, RangedClassifier classifier, FeatureCollection collection)
+            throws IOException {
+        double max = ((Number) classifier.getMax(classifier.getSize() - 1)).doubleValue();
+        double min = ((Number) classifier.getMin(0)).doubleValue();
+        double classWidth = (max - min) / classNum;
+        Subtract subtract = FF.subtract(getParameters().get(0), FF.literal(min));
+        Divide divide = FF.divide(subtract, FF.literal(classWidth));
+        Function abs = FF.function("abs", divide);
+        GroupByVisitor groupBy =
+                new GroupByVisitor(
+                        Aggregate.COUNT, getParameters().get(0), Arrays.asList(abs), null);
+        collection.accepts(groupBy, null);
+        Map<List<Integer>, Integer> result = groupBy.getResult().toMap();
+        return calculateNumericalPercentages(result, collection.size(), classNum);
+    }
+
+    private double[] calculateNumericalPercentages(
+            Map<List<Integer>, Integer> queryResult, int totalSize, int classNum) {
+        double[] percentages = new double[classNum];
+        Set<Integer> keys =
```

Comment:
So the treeset is used to make sure the classes are ordered from low to high right? And then collect the percentages in the right order? I'd add a comment about it.

Reply:
fixed